test/browser/lifecycles/componentWillMount.test.js

```diff
@@ -40,5 +43,66 @@ describe('Lifecycle methods', () => {
 
 			expect(componentState).to.deep.equal({ value: 1 });
 		});
+
+		it('should call setState callback in componentWillMount', () => {
+			let componentState;
+			let spy = sinon.spy();
+
+			class Foo extends Component {
+				constructor(props) {
+					super(props);
+					this.state = {
+						value: 0
+					};
+				}
+				componentWillMount() {
+					this.setState({ value: 1 }, spy);
+				}
+				render() {
+					componentState = this.state;
+					return <div />;
+				}
+			}
+
+			render(<Foo />, scratch);
+
+			expect(spy).to.be.calledOnce;
+			expect(componentState).to.deep.equal({ value: 1 });
+		});
+
+		it.skip('should call setState callback in correct order', () => {
```

Comment:
I don't think this is possible yet, since we call `diff` in `forceUpdate` and then `commitRoot` --> this function calls the `didMount`. in `diff` however we resolve the setState callbacks.